### DISCUSSION TITLE
Is private termination schedule actually required?

### DISCUSSION THREAD

**Discussion by alexaunter** [galactic3/ft-lockup-contract#7] [Q&A] [answered]
Current implementation allows to hide termination conditions (schedule) and reveal it only upon termination.

However, lockup schedule is NOT hidden, and if it's present and not trivial - it already gives off a lot of information about termination schedule.

Is there an existing private vesting use case that we are trying to replicate?
Should we also add ability to hide lockup schedule?

**Comment by AngelBlock** [comment]
Would be good to hear some use cases for the private vesting schedule, I was told it deprecated in existing lockup, what is the reason to reintroduce it in the new lockup logic?

**Comment by AngelBlock** [comment, marked as answer]
We agreed to implement / keep the private schedule.

**Reply by frol** [reply]
Private schedule is nice to have for the tokens that are not yet launched, but the promise of tokens allocation has already been given (e.g. a person started 1 year before the launch date and so he/she is eligible for the 25% of the tokens [assuming common 4 year lockup with 1 year cliff], yet, that would be a major liquidity bump if those tokens are immediately unlocked, so we are back to vesting + lock schedule).

I hope the difference of the new vs old lockup contracts regarding the vested vs locked tokens schedules will be that they re-use the same data structures and APIs, so it is much easier to reason about the final result.
